Document Dependency management and produceExecutableDuringBuild()/produceFeatureDuringBuild() methods (#842)
MicroEJ/docs: SDK6UserGuide/appendix/dependenciesManagement.rst

.. _gradle_dependencies_management_chapter:

> Dependencies management

Reviewer: Shouldn't it be "Dependencies Configurations" instead?

Author: I renamed the chapter.

=======================

This chapter describes all the dependency configurations added to your project by the MicroEJ Gradle plugins.

.. note::

>    The MicroEJ Gradle plugins extends the Gradle Java and Java Library plugins.

Reviewer: ```suggestion
   The MicroEJ Gradle plugins extend the Gradle Java and Java Library plugins.
```

   For more information about the configurations inherited from those plugins, 
   refer to the official documentation :
   
      - `Java plugin <https://docs.gradle.org/current/userguide/java_plugin.html#sec:java_plugin_and_dependency_management>`__
      - `Java Library plugin <https://docs.gradle.org/current/userguide/java_library_plugin.html#sec:java_library_separation>`__

The following diagrams show the dependency configurations. Use this legend to interpret the colors:

    - Green background : Dependencies can be declared against this configuration
    - Gray background : This configuration is for consumption by tasks only
    - Blue background : A task or plugin

.. graphviz:: graphConfigurations.dot

Publication Variants
--------------------

The :ref:`sdk6_module_natures.application` plugin defines a list of variants 
that are used during the publication of an Application.

microejWPK
~~~~~~~~~~

This variant is used to publish the WPK of an Application, 

> that can be fetched by declaring a dependency against the ``microejApplication`` configuration.

Reviewer: ```suggestion
that can be fetched by declaring a dependency with the ``microejApplication`` configuration.
```

Author: Done.

> The LibaryElement is set to ``microej-wpk``.

Reviewer: ```suggestion
The ``LibraryElement`` attribute of the variant is set to ``microej-wpk``.
```

Author: Done.

microejExecutable
~~~~~~~~~~~~~~~~~

This variant is used to publish the Executable of an Application, 

> that can be fetched by declaring a dependency against the ``microejKernel`` configuration.

Reviewer: ```suggestion
that can be fetched by declaring a dependency with the ``microejKernel`` configuration.
```

Author: Done.

> The LibaryElement is set to ``microej-executable``.

Reviewer: ```suggestion
The ``LibraryElement`` attribute of the variant is set to ``microej-executable``.
```

Author: Done.

microejExecutableBuildFiles
~~~~~~~~~~~~~~~~~~~~~~~~~~~

This variant is used to publish the files generated when building the Executable of an Application.

> The LibaryElement is set to ``microej-build-files``.

Reviewer: ```suggestion
The ``LibraryElement`` attribute of the variant is set to ``microej-build-files``.
```

Author: Done.

microejVirtualDevice
~~~~~~~~~~~~~~~~~~~~

This variant is used to publish the Virtual Device of an Application, 

> that can be fetched by declaring a dependency against the ``microejVeePort`` or the ``microejKernel`` configuration.

Reviewer: ```suggestion
that can be fetched by declaring a dependency with the ``microejVeePort`` or the ``microejKernel`` configuration.
```

> The LibaryElement is set to ``microej-vee-port``.

Reviewer: ```suggestion
The ``LibraryElement`` attribute of the variant is set to ``microej-vee-port``.
```

Author: Done.

..

>    | Copyright 2008-2023, MicroEJ Corp. Content in this space is free 

Reviewer: ```suggestion
   | Copyright 2008-2024, MicroEJ Corp. Content in this space is free 
```

Author: Done.
